src/Generating/FinishGen.cpp

```diff
@@ -708,8 +708,12 @@ bool cFinishGenSprinkleFoliage::TryAddSugarcane(cChunkDesc & a_ChunkDesc, int a_
 		return false;
 	}
 
-	// All conditions met, place a sugarcane here:
-	a_ChunkDesc.SetBlockType(a_RelX, a_RelY + 1, a_RelZ, E_BLOCK_SUGARCANE);
+	// All conditions met, place a sugarcane here
+	for (int i = 0; i < (a_RelX % m_MaxSugarcaneHeight); i++)
+	{
+		a_ChunkDesc.SetBlockType(a_RelX, a_RelY + i + 2, a_RelZ, E_BLOCK_SUGARCANE);
```

Comment:
I don't think `+ 2` is right here, won't the sugarcanes hover above the ground?

Reply:
Yes indeed, I'll change this